lib/ssz_ex.ex

```diff
@@ -151,6 +150,30 @@ defmodule LambdaEthereumConsensus.SszEx do
     end
   end
 
+  def hash_tree_root_list_composite_type(list, inner_schema, limit, len) do
+    root =
+      list
+      |> Enum.reduce(<<>>, fn value, acc_roots ->
+        acc_roots <> hash_tree_root!(value, inner_schema)
```

Comment:
This should use `hash_tree_root` as it could fail, right?

Reply:
What I mean is that `hash_tree_root!` can raise errors, so we should use `hash_tree_root` directly, and propagate errors.

Reply:
So, we return the error instead of it failing, right?

Reply:
Yes

Reply:
I don't think this should be the case here.

It shouldn't fail and this is inside of `Enum.reduce` so if it fails then there's probably an issue with the values or a bug.

Reply:
You can use `reduce_while` and an `{:ok, value}` accumulator to handle errors here. I understand this would fail only in rare circumstances, but we should keep `hash_tree_root` as an exception-free alternative to `hash_tree_root!`

Reply:
Okay. What is the ideal scenario to use `hash_tree_root!`?

Reply:
When we are sure that there won't be any errors.

Reply:
Updated it. Is this okay?

Reply:
Yes 👍